delocate/libsana.py

```diff
@@ -243,6 +244,140 @@ def walk_directory(
                 yield library_path
 
 
+def _tree_libs_from_libraries(
+    libraries: Iterable[str],
+    *,
+    lib_filt_func: Callable[[str], bool],
+    copy_filt_func: Callable[[str], bool],
+    executable_path: Optional[str] = None,
+    ignore_missing: bool = False,
+) -> Dict[str, Dict[str, str]]:
+    """ Return an analysis of the dependencies of ``libraries``.
+
+    libraries : iterable of str
+        The paths to the libraries to find dependencies of.
+    lib_filt_func : callable
+        A callable which accepts filename as argument and returns True if we
+        should inspect the file or False otherwise.
+        If `filt_func` filters a library it will will not further analyze any
+        of that library's dependencies.
+    copy_filt_func : None or callable
+        Called on each library name detected as a dependency; copy
+        where ``copy_filt_func(libname)`` is True, don't copy otherwise.
+    executable_path : None or str, optional
+        If not None, an alternative path to use for resolving
+        `@executable_path`.
+    ignore_missing : bool, default=False
```

Comment:
```suggestion
    ignore_missing : bool, default=False, optional
```

Reply:
The style guide is slightly vague here:

> If the default value would not be used as a value, optional is preferred.

I take this to mean that if the default value is not something you'd ever explicitly pass to the function, then use `optional` instead of `default`.  The style guide has no examples of both `default` and `optional` being used for the same parameter.  So you use `optional` when you want to ignore or hide the default value in the function description.